app/registries/foreman/plugin/importer_registry.rb

```diff
@@ -0,0 +1,25 @@
+module Foreman
+  class Plugin
+    class ImporterRegistry
+      attr_accessor :importers
+
+      def initialize
+        @importers = {}
+        @importers.default = 'PuppetFactImporter'
+      end
+
+      def register(key, klass, default = false)
+        @importers.default = klass if default
+        @importers[key.to_sym] = klass
+      end
+
+      def get(key)
+        @importers[key].constantize
+      end
+
+      def fact_features
+        importers.map { |_type, importer| importer.authorized_smart_proxy_features }.compact.flatten.uniq.constantize
+      end
+    end
+  end
+end
```

Comment:
I don't understand this - why plugins would create registries? Isn't this job of Foreman core to offer such a service?